File: jobjogger_api/app/models/notification.rb
# frozen_string_literal: true

class Notification < ApplicationRecord
  belongs_to :user
  belongs_to :job, optional: true

  enum :kind, {
    stage_stall:        0,
    deadline_reminder:  1,
    follow_up_due:      2,
    interview_reminder: 3
  }

  validates :body, presence: true
  validates :kind, presence: true

  scope :unread,  -> { where(read_at: nil) }
  scope :recent,  -> { order(created_at: :desc) }

  def mark_read!
    update!(read_at: Time.current)
  end

  def self.unread_count_for(user)
    where(user: user, read_at: nil).count
  end
end

File: jobjogger_web/src/types/notification.ts
export type NotificationKind =
  | 'stage_stall'
  | 'deadline_reminder'
  | 'follow_up_due'
  | 'interview_reminder'

export interface Notification {
  id: number
  kind: NotificationKind
  body: string
  read_at: string | null
  created_at: string
  job?: {
    id: number
    job_title: string
    company_name: string
  }
}

export interface NotificationsResponse {
  notifications: Notification[]
  meta: {
    unread_count: number
  }
}
